### PR TITLE
Update protobuf  version and solve unit tests

### LINKED ISSUE #3285: Upgrade to google.golang.org/protobuf 1.28.1 breaks unit test assertions
Upgrading from [google.golang.org/protobuf](http://google.golang.org/protobuf) 1.28.0 to 1.28.1 and results in some unit test failures. Something in this latest version of protobuf breaks some of the assertions in our code for types that embed protobuf messages as fields.

Example: [these assertions](https://github.com/spiffe/spire/blob/3e56c0120cff69f53e202e6fb858a60ef7cd514c/pkg/agent/manager/storecache/cache_test.go#L540-L541) in `pkg/agent/manager/storecache:TestUpdateEntries` fail, likely because the [storecache.Record](https://github.com/spiffe/spire/blob/3e56c0120cff69f53e202e6fb858a60ef7cd514c/pkg/agent/manager/storecache/cache.go#L21) and [cache.StaleEntry](https://github.com/spiffe/spire/blob/3e56c0120cff69f53e202e6fb858a60ef7cd514c/pkg/agent/manager/cache/cache.go#L126) structs include a *[common.RegistrationEntry](https://github.com/spiffe/spire/blob/3e56c0120cff69f53e202e6fb858a60ef7cd514c/proto/spire/common/common.proto#L59-L91) protobuf message field that is not cleanly matching using `reflect.DeepEqual()`. Digging into the failure, it looks like some internal proto field called `atomicMessageInfo` is not matching - one of the objects contains a pointer value for that field, the other a nil value.

It is probably worth evaluating whether protobuf messages should be embedded in Go structs in SPIRE code because of the challenges involved with comparing Go structs containing several fields, where some of the fields are plain Go structs and some are protobuf messages.

### PR DESCRIPTION
**Pull request by MarcosDY**
Upgrade protobuf  to 1.28.1 and solve test failures

**Which issue this PR fixes**
fixes #3285

### CODE REVIEW

File: pkg/agent/manager/storecache/cache_test.go [hunk 1]
```diff
@@ -351,8 +350,7 @@ func TestUpdateEntries(t *testing.T) {
 				},
 			},
 			setUpdate: func(update cache.UpdateEntries) *cache.UpdateEntries {
-				e := update.RegistrationEntries["foh"]
-				updatedEntry := proto.Clone(e).(*common.RegistrationEntry)
+				updatedEntry := createTestEntry()
```
Comment (azdagron) [inline pkg/agent/manager/storecache/cache_test.go:353]: Why do this instead of changing the require.Equal below to spiretest.AssertProtoListEqual?

**Review by azdagron — COMMENTED** [review]
Looks good! Just one comment.

File: pkg/agent/manager/storecache/cache_test.go (hunk 1 above)
Comment (MarcosDY) [reply pkg/agent/manager/storecache/cache_test.go:353]: equals below uses [Record ](https://github.com/MarcosDY/spire/blob/66cd401abae46bd9b4c7718c527344e88add820a/pkg/agent/manager/storecache/cache.go#L17) that are not an struct 
same happens with [staleentries](https://github.com/MarcosDY/spire/blob/66cd401abae46bd9b4c7718c527344e88add820a/pkg/agent/manager/cache/cache.go#L124)
And ` spiretest.AssertProtoListEqual` requires a proto.Message, 
the ideal solution for this is refactor our code to not longer cache proto messages in favor to structs, 
and we'll do that in a separate PR once changes on LRU are merged

**Comment by rturner3** [comment]
nit: we are actually upgrading `google.golang.org/protobuf` in this PR, not grpc-go

**Review by azdagron — COMMENTED** [review]
💯 on the move into the namedpipe package!!

**Review by azdagron — APPROVED** [review]

**Pull request merged**